qemu/lib/iozone.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import os
+
+from avocado.core import exceptions
+from virttest import utils_misc
+from virttest import postprocess_iozone
+
+
+class Iozone(object):
+    """
+    Prepare iozone env, run iozone on guest, clean up iozone env.
+    """
+    def __init__(self, test, params, vm, session):
+        """
+        run setup of iozone test
+        """
+        self.params = params
+        self.session = session
+        self.test = test
+        self.results_path = ""
+        self.timeout = float(params.get("iozone_timeout", 240))
+        check_cmd = params.get("check_cmd")
+        download_cmd = params.get("download_cmd")
+        setup_cmd = params.get("setup_cmd")
+        install_status = self.install_check(check_cmd)
+        if not install_status:
+            self.setup(download_cmd, setup_cmd)
+
+    def install_check(self, check_cmd):
+        """
+        check if iozone installed in guest
+        """
+        if check_cmd:
+            output = self.session.cmd_output(check_cmd)
+            if "iozone -h" in output:
+                return True
+            else:
+                return False
+
+    def setup(self, download_cmd, setup_cmd):
+        """
+        Download iozone and install it in guest
+        """
+        if download_cmd and setup_cmd:
+            logging.info("Download iozone tarball")
+            self.session.cmd(download_cmd, timeout=self.timeout)
+            logging.info("Setup iozone")
+            self.session.cmd(setup_cmd, timeout=self.timeout)
+
+    def run(self, test_cmd, mountpoint):
+        """
+        Run iozone test in guest
+
+        :param mountpoint: mountpoint/disk letter assigned to disk under test
+        :return: iozone run status and result.
+        """
+        #run iozone on the same device in parallel is not supported
+        test_cmd = test_cmd % mountpoint
+
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        if self.params.get("record_results", "yes"):
+            test_disk = mountpoint.split("/")[-1]
+            self.results_path = os.path.join(self.test.resultsdir,
+                                             'iozone_result_%s' % test_disk)
+            logging.debug("Write results to %s" % self.results_path)
+            try:
+                with open(self.results_path, 'a') as iozone_result:
+                    iozone_result.write(output)
+            except:
+                self.results_path = None
+                raise exceptions.TestFail("Failed to write iozone test results to file")
+
+            if self.params.get("post_results", "no") == "yes" and self.results_path:
```

Comment:
when `self.results_path = None`, this line will not be executed, because line 72 `raise exceptions.TestFail` terminates it.

Reply:
Yes, you're right. No need to check self.results_path.